lib/azure/armrest/resource_group_based_service.rb

```diff
@@ -187,6 +238,37 @@ def list_in_all_groups
 
         array
       end
+
+      def get_model_class(service_name)
+        case service_name.downcase
```

Comment:
We can simply maintain a hash table for the lookup instead of long list of case/when